plasmapy/atomic/ionization_state.py

```diff
@@ -0,0 +1,501 @@
+from typing import Dict, List, Optional, Tuple, Union
+
+import astropy.units as u
+import collections
+import numpy as np
+import numbers
+import warnings
+from plasmapy.atomic.particle_class import Particle
+from plasmapy.atomic.particle_input import particle_input
+from plasmapy.utils import (AtomicError, ChargeError, InvalidParticleError, check_quantity)
+
+State = collections.namedtuple(
+    'State', [
+        'integer_charge',
+        'ionic_fraction',
+        'ionic_symbol',
+    ])
```

Comment:
Ooh, we could probably use a 3.7 dataclass here! It's not necessary by any means, but I'd just like to note down the possibility. I've been itching to try those.

AND I HAD TO WRITE 3.5 COMPATIBLE PYTHON AT WORK, SO NO F-STRINGS :sob: 

So yeah, I think I'm gonna write something therapeutic in 3.7. :laughing: